compile/compile.go

```diff
@@ -100,6 +101,17 @@ func (c *Compiler) WithAsBundle(enabled bool) *Compiler {
 	return c
 }
 
+// WithPruneUnused will make rules defined on the same package as the
+// entrypoint, but that are not in the entrypoint set, to be ignored.
+//
+// Notably this includes functions (they can't be entrypoints) and causes
+// the built bundle to no longer be semantically equivalent to the bundle built
+// without wasm, or optimizations.}
```

Comment:
Might be worth mentioning this option is for -t plan just as much?

Reply:
There's no real equivalence there, I thought... but yeah as it stands, the "optimizations" part is also misleading. I'll update the comment. Thanks!

Reply:
Updated. WDYT?

Reply:
Looks great!